### PR TITLE
Add Jetpack Constants

### PR DESCRIPTION
**Pull request by kraftbj**
Add the constants settings page here instead of the JN companion plugin.

Right now, when asking folks to test things like beta blocks, there is no built-in way for beta testers to do this without manually adjusting config files.

### CODE REVIEW

File: jetpack-beta.php [hunk 1]
```diff
@@ -1176,6 +1178,80 @@ static function check_for_main_files( $source, $remote_source ) {
...
+		if ( ! defined( 'JETPACK__SANDBOX_DOMAIN' ) && self::constant_get_option( 'jetpack_sandbox_domain', '' ) ) {
+			define( 'JETPACK__SANDBOX_DOMAIN', self::constant_get_option( 'jetpack_sandbox_domain', '' ) );
+		}
+
+		if ( ! defined( 'JETPACK_PROTECT__API_HOST' ) && self::constant_get_option( 'jetpack_protect_api_host', '' ) ) {
+			define( 'JETPACK_PROTECT__API_HOST', self::constant_get_option( 'jetpack_protect_api_host', '' ) );
+		}
+		if ( ! defined( 'JETPACK_BETA_BLOCKS' ) && self::constant_get_option( 'jetpack_beta_blocks', '' ) ) {
+			define( 'JETPACK_BETA_BLOCKS', self::constant_get_option( 'jetpack_beta_blocks', '' ) ? true : false );
+		}
```
Comment (jeherve) [inline jetpack-beta.php:1254]: You seem to be missing some content here.

```suggestion
		}
	}

	/**
```

**Review by jeherve — CHANGES_REQUESTED** [review]
I like that idea!

Maybe that content should be found under Jetpack > Beta though?

**Comment by kraftbj** [comment]
Going to port this over to the monorepo FYI.

**Comment by kraftbj** [comment]
Closed in favor of https://github.com/Automattic/jetpack/pull/18537

**Pull request closed without merging**